ocpi-toolkit-2.2.1/src/main/kotlin/com/izivia/ocpi/toolkit/modules/credentials/services/CredentialsCommon.kt

```diff
@@ -0,0 +1,52 @@
+package com.izivia.ocpi.toolkit.modules.credentials.services
+
+import com.izivia.ocpi.toolkit.common.OcpiServerNoMatchingEndpointsException
+import com.izivia.ocpi.toolkit.modules.versions.domain.Endpoint
+import com.izivia.ocpi.toolkit.modules.versions.domain.InterfaceRole
+import com.izivia.ocpi.toolkit.modules.versions.domain.ModuleID
+
+class CredentialsCommon {
+    companion object {
+        fun checkRequiredEndpoints(
+            requiredEndpoints: Map<InterfaceRole, List<ModuleID>>,
+            actualEndpoints: List<Endpoint>
+        ) {
+            if (requiredEndpoints.isEmpty()) {
+                return
+            }
+
+            val requiredOtherPartAsReceiverEndpoints = requiredEndpoints[InterfaceRole.RECEIVER] ?: listOf()
+            val requiredOtherPartAsSenderEndpoints = requiredEndpoints[InterfaceRole.SENDER] ?: listOf()
+
+            actualEndpoints
+                .find { it.identifier == ModuleID.credentials }
+                .let {
+                    it ?: throw OcpiServerNoMatchingEndpointsException(
+                        "${ModuleID.credentials} other part endpoint missing"
+                    )
+                }
```

Comment:
I don't think you need the let ... just react to the null
```suggestion
               ?: throw OcpiServerNoMatchingEndpointsException(
                        "${ModuleID.credentials} other part endpoint missing"
                    )
               ```